### PR TITLE
Bugfix/noid/fix websocket connection for chat

### PR DESCRIPTION
**Pull request by mahibi**
### 🏁 Checklist

- [ ] ⛑️ Tests (unit and/or integration) are included or not needed
- [ ] 🔖 Capability is checked or not needed 
- [ ] 🔙 Backport requests are created or not needed: `/backport to stable-xx.x`
- [ ] 📅 Milestone is set
- [ ] 🌸 PR title is meaningful (if it should be in the changelog: is it meaningful to users?)

### CODE REVIEW

File: app/src/main/java/com/nextcloud/talk/webrtc/WebSocketInstance.kt [hunk 1]
```diff
@@ -391,7 +391,7 @@ class WebSocketInstance internal constructor(
                 roomToken == currentRoomToken &&
                 normalBackendSession == currentNormalBackendSession &&
                 federation?.roomId == currentFederation?.roomId &&
-                federation?.nextcloudServer == federation?.nextcloudServer
+                federation?.nextcloudServer == currentFederation?.nextcloudServer
```
Comment (danxuliu) [inline app/src/main/java/com/nextcloud/talk/webrtc/WebSocketInstance.kt:394]: :see_no_evil: What I was thinking on... Thanks!

File: app/src/main/java/com/nextcloud/talk/chat/ChatActivity.kt [hunk 2]
```diff
@@ -2514,57 +2503,78 @@ class ChatActivity :
...
-                    if (signalingSettingsOverall.ocs!!.settings!!.externalSignalingServer == null ||
-                        signalingSettingsOverall.ocs!!.settings!!.externalSignalingServer?.isEmpty() == true
-                    ) {
-                        return
+            )
+                .subscribeOn(Schedulers.io())
+                ?.observeOn(AndroidSchedulers.mainThread())
+                ?.subscribe(object : Observer<SignalingSettingsOverall> {
+                    override fun onSubscribe(d: Disposable) {
+                        disposables.add(d)
```
Comment (danxuliu) [inline app/src/main/java/com/nextcloud/talk/chat/ChatActivity.kt:2516]: Just to be sure, that means that if the signaling settings are received once the activity was closed `onNext` will not be executed (so the Websocket instance will not be got, not the room joined, nor the listeners added), is that correct? If not then that should be somehow prevented, but I guess that is exactly the purpose of the disposable :-)

**Review by danxuliu — APPROVED** [review]
I have not tested, but the changes make sense and look good :+1: 

Approved, as I guess that the disposable does what I expected it to do, but please check the comment before merging :-)

File: app/src/main/java/com/nextcloud/talk/chat/ChatActivity.kt (hunk 2 above)
Comment (mahibi) [reply app/src/main/java/com/nextcloud/talk/chat/ChatActivity.kt:2516]: yes that's the purpose. won't be necessary anymore soon when we switch from rxJava to coroutines

**Pull request merged**